pyvista/core/filters/data_set.py

```diff
@@ -6239,6 +6239,236 @@ def extract_cells_by_type(self, cell_types, progress_bar=False):
+        slower. For best performance, consider upgrading VTK.
+
+        Parameters
+        ----------
+        sort : bool, default: False
+            Whether to sort the output by label count in descending order
+            (i.e. from largest to smallest).
+
+        scalars : str, optional
+            Name of scalars to pack. Defaults to currently active scalars.
+
+        preference : str, default: "point"
+            When ``scalars`` is specified, this is the preferred array
+            type to search for in the dataset.  Must be either
+            ``'point'`` or ``'cell'``.
+
+        output_scalars : str, None
+            Name of the packed output scalars. By default, the output is
+            saved to currently active scalars.
+
+        progress_bar : bool, default: False
+            If ``True``, display a progress bar. Has no effect if VTK
+            version is lower than 9.3.
+
+        inplace : bool, default: False
+            If ``True``, the mesh is updated in-place.
+
+        Returns
+        -------
+        pyvista.Dataset
+            Dataset with packed labels.
+
+        Examples
+        --------
+        Pack segmented image labels.
+
+        Load non-contiguous image labels
+        >>> from pyvista import examples
+        >>> import numpy as np
+        >>> image_labels = examples.download_frog_tissue()
+
+        Show range of labels
+        >>> image_labels.get_data_range()
+        (0, 29)
+
+        Find 'gaps' in the labels
+        >>> label_numbers = np.unique(image_labels.active_scalars)
+        >>> label_max = np.max(label_numbers)
+        >>> missing_labels = set(range(label_max)) - set(label_numbers)
+        >>> len(missing_labels)
+        4
+
+        Pack labels to remove gaps
+        >>> packed_labels = image_labels.pack_labels()
+
+        Show range of packed labels
+        >>> packed_labels.get_data_range()
+        (0, 25)
+
+        """
+        # Determine input scalars
+        if scalars is None:
+            set_default_active_scalars(self)
+            _, scalars = self.active_scalars_info
+        arr = get_array(self, scalars, preference=preference, err=True)
+
+        field = get_array_association(self, scalars, preference=preference)
+
+        # Determine output scalars
+        if output_scalars is None:
+            output_scalars = scalars
+        elif isinstance(output_scalars, str):
+            output_scalars = output_scalars
+        else:
+            raise TypeError(f"Output scalars must be a string, got {type(output_scalars)} instead.")
+
+        # Do packing
+        if _vtk.VTK93:  # pragma: no cover
+            alg = _vtk.vtkPackLabels()
+            alg.SetInputDataObject(self)
+            alg.SetInputArrayToProcess(0, 0, 0, field.value, scalars)
+            if sort:
+                alg.SortByLabelCount()
+            # vtkPackLabels does not always pass data correctly and will sometimes
```

Comment:
Because of this choice to copy later, we have to have three mesh objects when `inplace=False`.  `self`, `output` and `result`.  Is there a way to avoid this?  My guess is that this filter could be used with large datasets.

Reply:
This was implemented a while ago and I cannot recall exactly what the issue was or if I wrote a test specifically to check for this issue (probably not). But I believe this was a bug which effectively made the filter act as if `inplace=True` by default, or some of the arrays present in the input would be missing from the output despite setting the `PassData` vars to `On`. I'll have to look into it. Definitely agree copies should be avoid if possible though.

Reply:
After some testing I found that the issue is that `vtkPackLabels` by default does not pass the input array through to the output. The implementation has been updated in [31bc44c](https://github.com/pyvista/pyvista/pull/4802/commits/31bc44cce20ce0ddb1056a1df48f8b22982b25cf) to remove the copying that was done before. Now, only the input array is copied through to the output.